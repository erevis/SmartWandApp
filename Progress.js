import {StyleSheet, View, Text} from 'react-native'
import * as React from 'react';

export function Progress() {
    return (
      <View style={styles.container}>
        <Text>Progress</Text>
      </View>
    );
  }

const styles = StyleSheet.create({
    container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    },
});
  