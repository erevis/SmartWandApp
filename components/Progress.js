import {StyleSheet, View, Text, TouchableOpacity} from 'react-native'
import * as React from 'react';
import {createNativeStackNavigator} from '@react-navigation/native-stack'
import { NavigationContainer } from '@react-navigation/native';
import { Overview } from '../routes/Overview';
import { Challenges } from '../routes/Challenges';
import { Achievements } from '../routes/Achievements';
import { COLORS } from './Colors';


const Stack = createNativeStackNavigator();

export function Progress() {
  const ref = React.useRef(null);
  
  function Default() {
    return(
      <View style={styles.container}>  
        <TouchableOpacity style={styles.button} onPress={() => {ref.current && ref.current.navigate('Overview')}}>
          <Text style={styles.text}>Overview</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.button} onPress={() => {ref.current && ref.current.navigate('Challenges')}}>
          <Text style={styles.text}>Challenges</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.button} onPress={() => {ref.current && ref.current.navigate('Achievements')}}>
          <Text style={styles.text}>Achievements</Text>
        </TouchableOpacity> 
      </View>
    );
  }

  return (
      <View style={{flex:1}}>
        <NavigationContainer independent={true} ref={ref}>
          <Stack.Navigator initialRouteName="Progress">
            <Stack.Screen name="Progress" component={Default}/>
            <Stack.Screen name="Overview" component={Overview}/>
            <Stack.Screen name="Challenges" component={Challenges}/>
            <Stack.Screen name="Achievements" component={Achievements}/>
          </Stack.Navigator>
        </NavigationContainer>  
      </View>
    )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.background,
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    margin: 10,
    padding: 10,
    backgroundColor: COLORS.color4,
    borderRadius:25,
    width:'50%',
    alignItems: 'center',
    justifyContent: 'center',
  },  
  text: {
    color: COLORS.background,
  }
});
