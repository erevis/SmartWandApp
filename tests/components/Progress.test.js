import React from 'react';
import renderer, { act } from 'react-test-renderer';

import {Progress} from '../../components/Progress.js';

describe('<Progress />', () => {
    act(() => {
        it('Renders correctly', () => {
            const tree = renderer.create(<Progress />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    })
})