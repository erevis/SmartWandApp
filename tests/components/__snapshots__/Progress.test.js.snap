// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> Renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        forwardedRef={[Function]}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="#202c40"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          hideShadow={false}
          largeTitleHideShadow={false}
          title="Progress"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": undefined,
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#476971",
                  "borderRadius": 25,
                  "justifyContent": "center",
                  "margin": 10,
                  "opacity": 1,
                  "padding": 10,
                  "width": "50%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": undefined,
                  }
                }
              >
                Overview
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#476971",
                  "borderRadius": 25,
                  "justifyContent": "center",
                  "margin": 10,
                  "opacity": 1,
                  "padding": 10,
                  "width": "50%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": undefined,
                  }
                }
              >
                Challenges
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#476971",
                  "borderRadius": 25,
                  "justifyContent": "center",
                  "margin": 10,
                  "opacity": 1,
                  "padding": 10,
                  "width": "50%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": undefined,
                  }
                }
              >
                Achievements
              </Text>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </RNCSafeAreaProvider>
</View>
`;
